tests/lib/tools/boot-state

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: bootenv show [var]"
+    echo "       bootenv set <var> <value>"
+    echo "       bootenv unset <var>"
+    echo "       boot-path"
+    echo "       wait-core-post-boot"
+    echo ""
+    echo "Get information and manages the boot loader for the current system"
+    echo ""
+    echo "COMMANDS:"
+    echo "  bootenv show:         prints the whole bootenv or just the variable passed as parameter"
+    echo "  bootenv set:          sets the given var and value on boot configuration"
+    echo "  bootenv unset:        unsets the given var from boot configuration"
+    echo "  boot-path:            prints the boot path"
+    echo "  wait-core-post-boot:  waits until the snap_mode bootenv var is empty"
+}
+
+get_grub_editenv() {
+    case "$SPREAD_SYSTEM" in
+        fedora-*|opensuse-*|amazon-*|centos-*)
+            command -v grub2-editenv
+            ;;
+        *)
+            command -v grub-editenv
+            ;;
+    esac
+}
+
+get_grub_envfile() {
+    case "$SPREAD_SYSTEM" in
```

Comment:
Could we do this instead?
```
if [ -e /boot/grub2/grubenv ]; then
    echo /boot/grub2/grubenv
else
    echo /boot/grub/grubenv
fi
```

This way we don't care about the system as much.

Reply:
nice, thanks